dipy/viz/actor.py

```diff
@@ -0,0 +1,648 @@
+    if (spline_subdiv is not None) and (spline_subdiv > 0):
+        spline_filter = set_input(vtk.vtkSplineFilter(), next_input)
+        spline_filter.SetSubdivideToSpecified()
+        spline_filter.SetNumberOfSubdivisions(spline_subdiv)
+        spline_filter.Update()
+        next_input = spline_filter.GetOutputPort()
+
+    poly_mapper = set_input(vtk.vtkPolyDataMapper(), next_input)
+    poly_mapper.ScalarVisibilityOn()
+    poly_mapper.SetScalarModeToUsePointFieldData()
+    poly_mapper.SelectColorArray("Colors")
+    # poly_mapper.SetColorModeToMapScalars()
+    poly_mapper.Update()
+
+    # Color Scale with a lookup table
+    if is_colormap:
+
+        if lookup_colormap is None:
+            lookup_colormap = colormap_lookup_table()
+
+        poly_mapper.SetLookupTable(lookup_colormap)
+        poly_mapper.UseLookupTableScalarRangeOn()
+        poly_mapper.Update()
+
+    # Set Actor
+    if lod:
+        actor = vtk.vtkLODActor()
+        actor.SetNumberOfCloudPoints(lod_points)
+        actor.GetProperty().SetPointSize(lod_points_size)
+    else:
+        actor = vtk.vtkActor()
+
+    # actor = vtk.vtkActor()
+    actor.SetMapper(poly_mapper)
+    actor.GetProperty().SetLineWidth(linewidth)
+    actor.GetProperty().SetOpacity(opacity)
+
+    return actor
+
+
+def lines_to_vtk_polydata(lines, colors=None):
+    """ Create a vtkPolyData with lines and colors
+
+    Parameters
+    ----------
+    lines : list
+        list of N curves represented as 2D ndarrays
+    colors : array (N, 3), list of arrays, tuple (3,), array (K,), None
+        If None then a standard orientation colormap is used for every line.
+        If one tuple of color is used. Then all streamlines will have the same
+        colour.
+        If an array (N, 3) is given, where N is equal to the number of lines.
+        Then every line is coloured with a different RGB color.
+        If a list of RGB arrays is given then every point of every line takes
+        a different color.
+        If an array (K, ) is given, where K is the number of points of all
+        lines then these are considered as the values to be used by the
+        colormap.
+        If an array (L, ) is given, where L is the number of streamlines then
+        these are considered as the values to be used by the colormap per
+        streamline.
+        If an array (X, Y, Z) or (X, Y, Z, 3) is given then the values for the
+        colormap are interpolated automatically using trilinear interpolation.
+
+    Returns
+    -------
+    poly_data :  vtkPolyData
+    is_colormap : bool, true if the input color array was a colormap
+    """
+
+    # Get the 3d points_array
+    points_array = np.vstack(lines)
+
+    nb_lines = len(lines)
+    nb_points = len(points_array)
+
+    lines_range = range(nb_lines)
+
+    # Get lines_array in vtk input format
+    lines_array = []
+    points_per_line = np.zeros([nb_lines], np.int64)
+    current_position = 0
+    for i in lines_range:
```

Comment:
You can iterate over lines instead. Then you wouldn't need to allocate `lines_range`